gettext-full: update to 0.22.5

PolynomialDivision opened the pull request.

mpratt14 commented:
there's so many changes to gettext between versions haha, like over 300 commits
changelog just notes major things

first thing I would try is to also update gnulib to a commit close to the latest, like this

https://git.savannah.gnu.org/cgit/gnulib.git/commit/?id=b90baa07983cc3e4eb53933de631296a5fc8c248

if that doesn't work we should look into removing the bootstrapping step, but we don't know the effect that would have on building with the SDK (whether it's "relocatable")

mpratt14 commented:
coreutils got a release

updating it and a minor tweak and it builds alongside the newer gnulib

https://github.com/mpratt14/openwrt/commit/934874f30ddbfbaabad11971050a78478acd235e

mpratt14 commented:
@PolynomialDivision I was thinking that you can pull that commit for coreutils into this PR because I tested it with this PR on top. 

coreutils may or may not build with the update without first updating gnulib because of the bootstrap step, a similar situation as gettext here.

but if you want I can pull your commit for gnulib and have the coreutils update in a separate PR, up to you

guidosarducci committed:
gettext-full: update to 0.22.5

Release Announcement:
https://savannah.gnu.org/news/?group_id=425

Refresh:
- 200-libunistring-missing-link.patch

Signed-off-by: Tony Ambardar <itugrok@yahoo.com>

PolynomialDivision commented:
@guidosarducci gettext fails:
```
 make[3] -C package/utils/checkpolicy host-compile
    ERROR: package/libs/gettext-full [host] failed to build.
make -r world: build failed. Please re-run make with -j1 V=s or V=sc for a higher verbosity level to see what's going on
make: *** [/__w/openwrt/openwrt/openwrt/include/toplevel.mk:233: world] Error 1
====== Make errors from logs/package/libs/gettext-full/host-compile.txt ======
lib/m4sugar/m4sh.m4:692: _AS_IF_ELSE is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
./lib/autoconf/general.m4:2249: AC_CACHE_VAL is expanded from...
./lib/autoconf/general.m4:2270: AC_CACHE_CHECK is expanded from...
aclocal.m4:6473: gt_INTDIV0 is expanded from...
../m4/intl.m4:39: gt_INTL_SUBDIR_CORE is expanded from...
configure.ac:105: the top level
configure:16703: error: possibly undefined macro: AM_INTL_SUBDIR
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
make[3]: *** [Makefile:194: /__w/openwrt/openwrt/openwrt/build_dir/hostpkg/gettext-0.22.5/.preparede6a00105302c492633b1fb38b6ed69cd_18f1e190c5d53547fed41a3eaa76e9e9] Error 1
time: package/libs/gettext-full/host-compile#176.61#56.52#260.51
```

mpratt14 commented:
it's affected by the version of gnulib used, just like almost everything else discussed in the other PR

PolynomialDivision commented:
But why should it be safe to merge this if it fails?

mpratt14 commented:
gnulib is updated in the other PR, you pulled the gettext update without the gnulib update here

guidosarducci commented:
Yet IIRC I had successfully built `gettext-full` before all the extra `gnulib` updates. If anything, I think we underestimate the impact of how entangled `gnulib` has become in the distro and how fragile it can render routine maintenance.

@PolynomialDivision Thanks for merging my dwarves maintainer patch earlier and at least trying with this one.
